x/derivatives/keeper/perpetual_futures.go

```diff
@@ -77,54 +77,28 @@ func (k Keeper) ClosePerpetualFuturesPosition(ctx sdk.Context, position types.Po
 	feeAmount := positionInstance.Size_.Mul(commissionRate)
 	tradeAmount := positionInstance.Size_.Sub(feeAmount)
 
-	// todo: fixme
-	openedRate := position.OpenedBaseRate
 	closedRate, err := k.GetPairRate(ctx, position.Market)
 	if err != nil {
 		return err
 	}
 
-	k.bankKeeper.SendCoinsFromAccountToModule(ctx, position.Address.AccAddress(), types.ModuleName, sdk.Coins{sdk.NewCoin(position.Market.BaseDenom, feeAmount.RoundInt())}) // TODO: this is wrong.
-
-	principal := positionInstance.CalculatePrincipal()
-	amountToUser := sdk.Dec{}
-
-	switch positionInstance.PositionType {
-	case types.PositionType_LONG:
-		k.SubPerpetualFuturesNetPositionOfMarket(ctx, position.Market, tradeAmount)
-
-		if closedRate.GTE(openedRate) {
-			profit := closedRate.Mul(sdk.NewDec(int64(positionInstance.Leverage))).Sub(positionInstance.Size_)
-			profitAmount := profit.Quo(*closedRate)
-
-			amountToUser = principal.Add(profitAmount)
-		} else {
-			loss := positionInstance.Size_.Sub(closedRate.Mul(sdk.NewDec(int64(positionInstance.Leverage))))
-			lossAmount := loss.Quo(*closedRate)
-
-			amountToUser = principal.Sub(lossAmount)
-		}
-		break
-	case types.PositionType_SHORT:
-		k.AddPerpetualFuturesNetPositionOfMarket(ctx, position.Market, tradeAmount)
-
-		if closedRate.LTE(openedRate) {
-			profit := positionInstance.Size_.Sub(closedRate.Mul(sdk.NewDec(int64(positionInstance.Leverage))))
-			profitAmount := profit.Quo(*closedRate)
-
-			amountToUser = principal.Add(profitAmount)
-		} else {
-			loss := closedRate.Mul(sdk.NewDec(int64(positionInstance.Leverage))).Sub(positionInstance.Size_)
-			lossAmount := loss.Quo(*closedRate)
+	err = k.bankKeeper.SendCoinsFromAccountToModule(ctx, position.Address.AccAddress(), types.ModuleName, sdk.Coins{sdk.NewCoin(position.Market.BaseDenom, feeAmount.RoundInt())}) // TODO: this is wrong.
```

Comment:
@taiki1frsh 
What happens if the owner of position has a zero balance?
Wouldn't it be better to take the commission from remaing margin?

Reply:
i didn't touch this.
i just added error case handling.
should i care this in this PR?

Reply:
oh, ok.
plz make issuen and write comment.

Reply:
This PR does not fix that.